tests/test_cli_compile.py

```diff
@@ -2494,9 +2497,275 @@ def test_all_extras_fail_with_extra(fake_dists, runner, make_module, fname, cont
     assert exp in out.stderr
 
 
+def _parse_deps(text: str, keys: Iterable[str]) -> dict[str, str | None]:
+    """Parse ``pip-compile`` output and return mapping from package name to package version
+
+    :param text: output from ``pip-compile``
+    :param keys: names of expected packages. If not in text the version will be set to ``None``.
+    """
+    result: dict[str, str | None] = {k: None for k in keys}
+
+    for line in text.splitlines():
+        prefix = line.split("#")[0].strip()
+        if not prefix:
+            continue
+        try:
+            k, v = prefix.split("==")
+        except ValueError:
+            continue
+        result[k] = v
```

Comment:
This could be simpler:
```suggestion
        pkg_spec = line.split("#")[0].strip()
        pkg_name, eq_sep, pkg_version = pkg_spec.partition('==')

        if not eq_sep:
            continue

        result[pkg_name] = pkg_value
```

Reply:
Good point.